### PR TITLE
allow reading build date from a BUILDDATE file

### PR DESCRIPTION
**Pull request by kib**
Could be used for environments where date information is incorrect (like Launchpad)
```
Build: Kodi 18.0-ALPHA3 Git: 20121104-6fcbb13
Compiled: Nov 4 2012
```
which was built on 20180626 

## Motivation and Context
Stops people from asking why the date is wrong.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: cmake/scripts/common/Macros.cmake [hunk 1]
```diff
@@ -662,7 +662,11 @@ function(core_find_git_rev stamp)
       string(REPLACE "\"" "" DATE ${DATE})
       string(REPLACE "-" "" DATE ${DATE})
     else()
-      string(TIMESTAMP DATE "%Y%m%d" UTC)
+      if(EXISTS ${CMAKE_SOURCE_DIR}/BUILDDATE)
+        file(STRINGS ${CMAKE_SOURCE_DIR}/BUILDDATE DATE LIMIT_INPUT 16)
```
Comment (Rechi) [inline cmake/scripts/common/Macros.cmake]: `LIMIT_INPUT 8` is enough for YYYYmmdd

**Comment by wsnipex** [comment]
thanks, looks good part from the limit rechi mentioned

File: cmake/scripts/common/Macros.cmake (hunk 1 above)
Comment (kib) [reply cmake/scripts/common/Macros.cmake]: That does make a lot of sense. WIll force push in a sec.

**Comment by kib** [comment]
@wsnipex should be ok now.

**Review by Rechi — APPROVED** [review]

**Comment by ccope** [comment]
:tada: towards reproducibility!

**Pull request merged**